### DISCUSSION TITLE
Getting the Project of a Pull Request

### DISCUSSION THREAD

**Discussion by AceCoderLaura** [octokit/octokit.js#2048] [Q&A]
Hi all, I'm building a GitHub Action to validate my pull requests but I'm having trouble checking if the PR is in a project. The project assigned to the PR doesn't appear anywhere in the PRs schema when retrieved from the REST service.

In my below code you can see what I'm trying to achieve. The first check works, the second check does not since there's no "projects" field on the returned object.

```ts
            const {data: pr} = await github.pulls.get({
              pull_number: context.payload.pull_request.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
            })

            if (pr.requested_reviewers.count < 1)
            {
              core.setFailed("You must assign someone to review a PR")
            }

            if (!pr.projects.count < 1)
            {
              core.setFailed("You must assign all PRs to a project")
            }
```

Any and all help is appreciated.

**Comment by wolfy1339** [comment]
There isn't currently an API to do what you would like to.

Can you use labels to assign a project? If so, the `octokit` team uses an app made to categorise PRs and issues based on which label is assigned to the PR/issue. You can see how it's handled in this code: https://glitch.com/edit/#!/octokit-js-project-board?path=index.js%3A3%3A45

**Reply by AceCoderLaura** [reply]
Probably wouldn't do well as an Action since everyone who installs the Action would have to implement the same workflow.

Since the API doesn't support it I won't try and force it. Thanks for the suggestion though.

**Reply by oscard0m** [reply]
@AceCoderLaura maybe in GitHub Community you can find other developers facing your problem or you can post there your problems. This seems to be a limitation in GitHub API for a Pull Request. It would have sense to get the list of linked Projects to a PR.

**Reply by gr2m** [reply]
This is the forum to ask about GitHub Platform API: https://github.community/c/github-ecosystem/37

**Comment by gr2m** [comment]
I think it's possible using `graphql`

```graphql
query { 
  resource(url:"https://github.com/gr2m/sandbox/pull/177") {
    ... on PullRequest {
      projectCards {
        nodes {
          note
          column {
            project {
              databaseId
              url
            }
          }
        }
      }
    }
  }
}
```

result looks like this

```json
{
  "data": {
    "resource": {
      "projectCards": {
        "nodes": [
          {
            "note": null,
            "column": {
              "project": {
                "databaseId": 2178975,
                "url": "https://github.com/gr2m/sandbox/projects/1"
              }
            }
          }
        ]
      }
    }
  }
}
```

You can send GraphQL queries using `octokit.graphql(query /* , variables */)`, the method is documented at https://github.com/octokit/graphql.js#readme

You can experiment creating your query at https://docs.github.com/en/graphql/overview/explorer
